doc/doc/advanced/pgxl.md

```diff
@@ -0,0 +1,310 @@
+key that is **not initially assigned**.
+
+Refer to the [XL
+manual](http://files.postgres-xl.org/documentation/) for
+more information.
+
+You should always create tables as `CREATE UNLOGGED TABLE ...`. The unlogged
+keyword turns off write-ahead logging (WAL), and gives a significant speedup
+in writes (often 10X). WAL is not needed for DeepDive apps, because the
+database is only used for processing, not for persisting data.
+
+## Installation
+We now describe how to install XL and configure it to be used with
+DeepDive. The steps were tested to install XL on Ubuntu 15.04. 
+
+### Setting OS Parameters
+
+Set the following parameters in the `/etc/sysctl.conf` file: 
+```
+kernel.sem = 1000  32000  32  1000
+kernel.shmmax = 429496729600  #400GB
+```
+
+After making these changes, run
+```bash
+sudo sysctl -p /etc/sysctl.conf
+```
+
+Set the following parameter in the `/etc/ssh/ssh_config` file:
+```
+StrictHostKeyChecking no
+```
+
+Finally, set the following parameter in the `/etc/ssh/sshd_config` file:
+```
+MaxStartups 100
+```
+After that, run
+```bash
+sudo service ssh restart
+```
+
+### Building XL from Source
+
+```bash
+sudo apt-get update
+sudo apt-get -y install -y screen curl git rsync openssl locales openssh-server openssh-client
+sudo apt-get -y install -y gcc flex bison make cmake jade openjade docbook docbook-dsssl
+sudo apt-get -y install zlib1g-dev libreadline6-dev python-dev libssl-dev
+sudo localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+```
+
+Now set `$TARGET_DIR` to the directory into which you would like to install XL.
+```bash
+TARGET_DIR=/opt/pgxl
+```
+
+Then add the following line(s) to `/etc/ld.so.conf` (replace `$TARGET_DIR` with dir):
+```bash
+/usr/local/lib
+$TARGET_DIR/lib
+```
+
+The remaining parts of the installation do not require sudo rights. However, make sure
+the user account doing the installation has write permissions to `$TARGET_DIR`.
+
+Create a build directory and download the XL sources.
+```bash
+BUILD_DIR=~/pgxl_install
+mkdir -p $BUILD_DIR
+cd $BUILD_DIR
+wget -O pgxl-v9.2.tar.gz http://sourceforge.net/projects/postgres-xl/files/Releases/Version_9.2rc/postgres-xl-v9.2-src.tar.gz/download
+tar -xzf pgxl-v9.2.tar.gz
+```
+
+Now, you can build the sources.
+```bash
+cd $BUILD_DIR/postgres-xl/
+./configure --with-python --with-openssl --prefix $TARGET_DIR
+make -j 4
+make install
+```
+
+Additionally modules can be added as follows.
+```bash
+for pkg in btree_gin btree_gist earthdistance fuzzystrmatch hstore intagg intarray oid2name \
+pg_buffercache pgcrypto pgxc_clean pgxc_ctl pgxc_ddl pgxc_monitor stormstats \
+tablefunc tsearch2 unaccent; do
+    cd $BUILD_DIR/postgres-xl/contrib/$pkg
+    make; make install
+done
+```
+
+### Set XL related session variables
+
+To set Greenplum related session variables, modify your
+`~/.bashrc` script and add the line:
+
+```bash
+export PATH=$TARGET_DIR/bin:\$PATH
```

Comment:
There shouldn't be a backslash before the last `$PATH`.  Surrounding quote may be safer.
```bash
export PATH="$TARGET_DIR/bin:$PATH"
```